code/modules/antagonists/eldritch_cult/eldritch_antag.dm

```diff
@@ -0,0 +1,249 @@
+	give_equipment = FALSE
+	GLOB.reality_smash_track.Generate(1)
+	new_owner.add_antag_datum(src)
+	message_admins("[key_name_admin(admin)] has heresized [key_name_admin(new_owner)].")
+	log_admin("[key_name(admin)] has heresized [key_name(new_owner)].")
+
+
+/datum/antagonist/heretic/greet()
+	owner.current.playsound_local(get_turf(owner.current), 'sound/ambience/antag/ecult_op.ogg', 100, FALSE, pressure_affected = FALSE)//subject to change
+	to_chat(owner, "<span class='boldannounce'>You are the Heretic!</span><br>\
+	<B>The old ones gave you these tasks to fulfill:</B>")
+	owner.announce_objectives()
+	to_chat(owner, "<span class='cult'>The book whispers, the forbidden knowledge walks once again!<br>\
+	Your book allows you to research abilities, read it very carefully! you cannot undo what has been done!<br>\
+	You gain charges by either collecitng influences or sacrifcing people tracked by the living heart<br> \
+	You can find a basic guide at :https://tgstation13.org/wiki/Heresy_101 </span>")
+
+/datum/antagonist/heretic/on_gain()
+	var/mob/living/current = owner.current
+	if(ishuman(current))
+		forge_primary_objectives()
+		gain_knowledge(/datum/eldritch_knowledge/spell/basic)
+		gain_knowledge(/datum/eldritch_knowledge/living_heart)
+	current.log_message("has been converted to the cult of the forgotten ones!", LOG_ATTACK, color="#960000")
+	GLOB.reality_smash_track.AddMind(owner)
+	START_PROCESSING(SSprocessing,src)
+	if(give_equipment)
+		equip_cultist()
+	return ..()
+
+/datum/antagonist/heretic/on_removal()
+
+	for(var/X in researched_knowledge)
+		var/datum/eldritch_knowledge/EK = X
+		EK.on_lose(owner.current)
+
+	if(!silent)
+		owner.current.visible_message("<span class='deconversion_message'>[owner.current] looks like [owner.current.p_theyve()] just reverted to [owner.current.p_their()] old faith!</span>", null, null, null, owner.current)
+		to_chat(owner.current, "<span class='userdanger'>Your mind begins to flare as the otherwordly knowledge escapes your grasp!</span>")
+		owner.current.log_message("has renounced the cult of the old ones!", LOG_ATTACK, color="#960000")
+	GLOB.reality_smash_track.RemoveMind(owner)
+	STOP_PROCESSING(SSprocessing,src)
+
+	return ..()
+
+
+/datum/antagonist/heretic/proc/equip_cultist()
+	var/mob/living/carbon/H = owner.current
+	if(!istype(H))
+		return
+	. += ecult_give_item(/obj/item/forbidden_book, H)
+	. += ecult_give_item(/obj/item/living_heart, H)
+
+/datum/antagonist/heretic/proc/ecult_give_item(obj/item/item_path, mob/living/carbon/human/H)
+	var/list/slots = list(
+		"backpack" = ITEM_SLOT_BACKPACK,
+		"left pocket" = ITEM_SLOT_LPOCKET,
+		"right pocket" = ITEM_SLOT_RPOCKET
+	)
+
+	var/T = new item_path(H)
+	var/item_name = initial(item_path.name)
+	var/where = H.equip_in_one_of_slots(T, slots)
+	if(!where)
+		to_chat(H, "<span class='userdanger'>Unfortunately, you weren't able to get a [item_name]. This is very bad and you should adminhelp immediately (press F1).</span>")
```

Comment:
Admins don't debug. Use an informative `stack_trace()` instead.

Reply:
this is required to be mentioned to player , since they werent given a very very important item, and they should ahelp for admins to spawn it manually, sometimes happens